ast/parser.go

```diff
@@ -361,61 +366,47 @@ func (p *Parser) parseElse(head *Head) *Rule {
 
 	var rule Rule
 	rule.SetLoc(p.s.Loc())
+
+	rule.Head = head.Copy()
+	setLocRecursive(rule.Head, p.s.Loc())
```

Comment:
This sets the location on the head, args, and value of the else block which then get changed below. I'm wondering if it would be clearer if this was removed or moved and called in the places where it's actually needed--for example:

* it must be called on the head itself to set the location of that node
* it must be called on the rule args because those cannot be redefined today but the locations should be updated
* it must be called if the value is omitted (the first case below)
* it does not have to be called if value is not omitted (the second case below)

Thoughts?

Reply:
Yep, I agree. Its kind of related to the issue described on the thread above this one with the chained rule head's.

> it must be called on the rule args because those cannot be redefined today but the locations should be updated

I'm not sure I follow that. Shouldn't their individual locations match up with their actual location? Why set them to match the full rule head?

Reply:
The latest update changes to keeping the original location for the arguments. The more I thought about it the more I'm convincing myself thats the right behavior.